From 94f86ffadee98ba7d1d41112ad816a51b5708b65 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 20 Jun 2013 20:44:44 +0530
Subject: [PORT FROM MAIN]AudioFlinger: check the mOutput before using it in Track::stop()

BZ: 66362

mOutput may be set to NULL in closeOutput() and accessed in
PlaybackThread::Track::stop(). Since the two accesses have been
protected by mLock, just add a check of mOutput before using it.

Change-Id: I886c71964e2402f2d01ed504f5f9122b2911d82c
Orig-Change-Id: I92eb38374516d0c6ec8d8e6808a1548075d39094
Signed-off-by: Zhi Jin <zhi.jin@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/Tracks.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/services/audioflinger/Tracks.cpp b/services/audioflinger/Tracks.cpp
index 7e3940c..0100d6f 100644
--- a/services/audioflinger/Tracks.cpp
+++ b/services/audioflinger/Tracks.cpp
@@ -705,7 +705,7 @@ void AudioFlinger::PlaybackThread::Track::stop()
 #ifdef INTEL_MUSIC_OFFLOAD_FEATURE
         PlaybackThread* playbackThread = static_cast<PlaybackThread*>(thread.get());
         if (isOffloadTrack()) {
-            if (state!=ACTIVE && state!=RESUMING) {
+            if (state!=ACTIVE && state!=RESUMING && playbackThread->getOutput_l()) {
                 ALOGV("Track:stop: offload state!=ACTIVE && state!=RESUMING");
                 status_t status = playbackThread->getOutput_l()->stream->flush(
                                            playbackThread->getOutput_l()->stream);
-- 
1.7.4.1

